azure/pipelines/dotnet/ci.yml

```diff
@@ -212,13 +202,31 @@ steps:
           exit 0;
 
   - ${{ if eq(parameters.AzureSubscription, '') }}:
-    - task: DotNetCoreCLI@2
-      displayName: 'DotNet Test'
+    # We don't use DotNetCoreCLI@2 for testing as it doesn't properly support nested solution folders in combination with MTP.
+    - task: PowerShell@2
+      displayName: DotNet Test
       inputs:
-        command: 'test'
-        projects: ${{ parameters.TestProjectsGlob }}
-        arguments: '--configuration ${{ parameters.BuildConfiguration }} ${{ parameters.PostBuildDotnetArgs }} ${{ parameters.TestCoverageArguments}} ${{ parameters.TestArguments }}'
+        targetType: inline
+        pwsh: true
+        script: |
+          # Sets test coverage default when not provided, defaults depends whether native MTP is configured with .NET 10 or not.
+          # Use '--coverage' when using native MTP, '--collect "Code Coverage"' for MTP over VSTest
+          $solutionDir = (Get-Item "${{ parameters.Solution }}").Directory
+          $isMtp = (Get-Content (Join-Path $solutionDir "global.json") -Raw -ErrorAction SilentlyContinue | ConvertFrom-Json).test.runner -eq "Microsoft.Testing.Platform"
 
+          Resolve-Path ${{ parameters.TestProjectsGlob }} -ErrorAction SilentlyContinue | Get-Item | Foreach-Object {
+            if ( $isMtp ) {
+              # Change working directory because global.json is expected in the working directory by 'dotnet test'.
+              Set-Location -Path $_.Directory
```

Comment:
@WimVergouwe Hmmm,, this sets the working directory to the containing folder of the 'csproj' file with '$_.Directory'.
Not sure why this works though.

Wouldn't 'Set-Location -Path $solutionDir' be better.

Reply:
Updated and tested